apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ExecuteHelper.cs

```diff
@@ -33,7 +34,7 @@ public static Task<T> WithSessionChecking<T>(this Task<T> task, Func<Session> se
                         SessionPool.MarkSessionExpired(sessionFunc());
                     }
                 }
-                return t.Result;
+                return t.ResultWithUnwrappedExceptions();
```

Comment:
Why is the `.ContinueWith` required? Could `WithSessionChecked` be implemented like this:
``` C#
try
{
    return await task.ConfigureAwait(false);
}
catch (RpcException ex)
{
    if (ex.IsSessionExpiredError()
    {
        SessionPool.MarkSessionExpired(sessionFunc());
    }
}
```